### PR TITLE
[ADLA - ADLS] - Object refactoring

### PR DESCRIPTION
**Pull request by ro-joowan**
* Based on this Swagger PR: Azure/azure-rest-api-specs#1452

* Based on this .NET PR: https://github.com/Azure/azure-sdk-for-net/pull/3512

* Updated the tests ever so slightly to make them more reliable.  I also ran them live and updated the SessionRecords folder for both ADLA and ADLS

* Besides code refactoring for type changes, the only real change is updating adapting to how Job submission/build works in src\ResourceManager\DataLakeAnalytics\Commands.DataLakeAnalytics\Commands\SubmitAzureRmDataLakeAnalyticsJob.cs

* For the deleted PS(insert some properties object).cs files, it is because via Swagger these properties have been flattened.  E.g. in src\ResourceManager\DataLakeAnalytics\Commands.DataLakeAnalytics\Models\PSDataLakeStoreAccountInfo.cs:

[Obsolete("In a future release this object will have all 'Properties' properties flattened and the 'Properties' property will be removed. Until then, nested properies will be duplicated.")]

* Updated the changelog for ADLA and ADLS

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by cormacpayne** [comment]
@ro-joowan it looks like there are some previously documented breaking changes being made in this release (_e.g.,_ removing the `Properties` property). These will need to be made for the November release rather than the October release. If anything needs to be made in the October release, I would recommend opening a separate PR with those changes so they can get in.

**Comment by ro-joowan** [comment]
@cormacpayne, This can go in the November release.  Thank you!

**Comment by ro-joowan** [comment]
@cormacpayne, Can we move forward from the "Do Not Merge" state?

**Comment by cormacpayne** [comment]
@azuresdkci test this please

File: src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/Commands.DataLakeAnalytics.csproj [hunk 1]
```diff
@@ -38,7 +38,19 @@
   <ItemGroup>
     <Reference Include="Microsoft.Azure.Management.DataLake.Analytics, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.DataLake.Analytics.3.0.0\lib\net452\Microsoft.Azure.Management.DataLake.Analytics.dll</HintPath>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.DataLake.Analytics.3.1.2-preview\lib\net452\Microsoft.Azure.Management.DataLake.Analytics.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/Commands.DataLakeAnalytics.csproj]: @ro-joowan please remove the references you are adding to `Microsoft.Rest.ClientRuntime`, `Microsoft.Rest.ClientRuntime.Azure` and `Newtonsoft.Json` in this file. These references are already covered in the `Common.Dependencies.targets` file in the `tools` folder.

File: src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/packages.config [hunk 2]
```diff
@@ -1,4 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Azure.Management.DataLake.Analytics" version="3.0.0" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.DataLake.Analytics" version="3.1.2-preview" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.10" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/packages.config]: @ro-joowan similar comment about remove the new references in this file. These references can be found in the `packages.config` file in the `Commands.Common.ResourceManager` project.

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands.DataLakeStore.csproj [hunk 3]
```diff
@@ -38,7 +38,19 @@
   <ItemGroup>
     <Reference Include="Microsoft.Azure.Management.DataLake.Store, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.DataLake.Store.2.2.0\lib\net452\Microsoft.Azure.Management.DataLake.Store.dll</HintPath>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.DataLake.Store.2.3.0-preview\lib\net452\Microsoft.Azure.Management.DataLake.Store.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands.DataLakeStore.csproj]: @ro-joowan same comment about removing the references in this file

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/packages.config [hunk 4]
```diff
@@ -1,4 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Azure.Management.DataLake.Store" version="2.2.0" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.DataLake.Store" version="2.3.0-preview" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/packages.config]: @ro-joowan same comment about removing the references in this file

File: src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/Commands.DataLakeAnalytics.csproj (hunk 1 above)
Comment (ro-joowan) [reply src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/Commands.DataLakeAnalytics.csproj]: Done - thanks!

File: src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/packages.config (hunk 2 above)
Comment (ro-joowan) [reply src/ResourceManager/DataLakeAnalytics/Commands.DataLakeAnalytics/packages.config]: Done - thanks!

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands.DataLakeStore.csproj (hunk 3 above)
Comment (ro-joowan) [reply src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands.DataLakeStore.csproj]: Done - thanks!

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/packages.config (hunk 4 above)
Comment (ro-joowan) [reply src/ResourceManager/DataLakeStore/Commands.DataLakeStore/packages.config]: Done - thanks!

**Comment by ro-joowan** [comment]
@cormacpayne - Changes made.  Thank you for that!

**Comment by cormacpayne** [comment]
@ro-joowan the build is failing because StaticAnalysis caught the breaking changes you are trying to make, so you will need to suppress these exception. To do so, you will need to use a text editor (Notepad, VS Code, Sublime, etc.) and open the `BreakingChangeIssues.csv` file found in the `Build Artifacts` of the job, copy the contents (except for the header) and append them to the end of `tools/StaticAnalysis/Exceptions/BreakingChangeIssues.csv` using a text editor. Let me know if you have any questions about this.

**Comment by ro-joowan** [comment]
@cormacpayne The build passed after making the update to the BreakingChangeIssues.csv.  Thanks!

**Pull request merged**